docs/0.1/grid_product_smart_contract_specification.md

```diff
@@ -0,0 +1,288 @@
+smart contract engine.
+
+Grid Product is designed to supply Hyperledger Grid with a generic and
+extendable framework for storing product master data. A product in this case, is
+any  item that is transacted, traded or referenced in a supply chain. Grid
+Product offers explicit support for products defined using the
+[GS1](https://www.gs1.org), but is also designed to be flexible enough to
+support other standards.
+
+Grid Product uses the Pike smart contract to manage permissions for creating,
+updating, and deleting products, as well as for managing the organizations
+that own each product, and uses the Schema smart contract to describe the
+requirements for the product entities.
+
+## State
+
+All Grid Product object are serialized using Protocol Buffers before being
+stored in state. These objects include Product, `ProductList`, and
+`PropertyValue`.
+
+### Product
+
+A product is an arbitrary list of properties that is uniquely identified by a
+`product_id`. The `properties` are described by a schema that was previously
+defined using the Schema smart contract. The schema that the product will be
+checked against is dictated by the product's `product_namespace` field.
+Currently, only the `GS1` namespace has been defined.
+
+A product has four fields
+
+* product_id: A unique identifier for the product. For the GS1 namespace the
+  product_id is a GS1 GTIN.
+* product_namespace: The namespace that the product belongs to. A product's
+  namespace dictates the rules for defining its `product_id` and the
+  properties belonging to the product. Currently only the GS1 namespace is
+  defined.
+* owner: The Pike organization ID of the organization that owns the product.
+* properties: An arbitrary list of properties that describes the product. These
+  properties are defined by a schema defined by the Schema smart contract.
+
+```protobuf
+    message Product {
+        enum ProductNamespace {
+            UNSET_NAMESPACE = 0;
+            GS1 = 1;
+        }
+
+        ProductNamespace product_namespace = 1;
+        string product_id = 2;
+        string owner = 3;
+        repeated PropertyValue properties = 4;
+    }
+```
+
+### Product List
+
+Products whose addresses collide are stored in a product list. A product list
+contains only one field.
+
+```protobuf
+    message ProductList {
+      repeated Product entries = 1;
+    }
+```
+
+### Addressing for GS1 products
+
+In order to uniquely locate GS1 products in the Merkle-Radix state system, an
+address must be constructed which identifies the storage location of the Product
+representation.
+
+All Grid addresses are prefixed by the 6-hex-character namespace prefix
+“621dee”,  Products are further prefixed under the Grid namespace with reserved
+enumerations of “02” (“00” and “01” being reserved for other purposes)
+indicating “Products” and an additional “01” indicating “GS1 Products”.
+
+Therefore, all addresses starting with:
```

Comment:
I think this sentence is incomplete. Maybe something like "Therefore, addresses with starting with the following string are Grid products"